jsettlers.graphics/src/jsettlers/graphics/map/controls/original/panel/content/DistributionPanel.java

```diff
@@ -0,0 +1,286 @@
+/*******************************************************************************
+ * Copyright (c) 2015
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *******************************************************************************/
+package jsettlers.graphics.map.controls.original.panel.content;
+
+import go.graphics.text.EFontSize;
+import jsettlers.common.buildings.EBuildingType;
+import jsettlers.common.images.EImageLinkType;
+import jsettlers.common.images.ImageLink;
+import jsettlers.common.images.OriginalImageLink;
+import jsettlers.common.map.IGraphicsGrid;
+import jsettlers.common.position.ShortPoint2D;
+import jsettlers.graphics.action.Action;
+import jsettlers.graphics.action.ExecutableAction;
+import jsettlers.graphics.localization.Labels;
+import jsettlers.graphics.map.controls.original.panel.content.Label.HorizontalAlignment;
+import jsettlers.graphics.utils.UIPanel;
+
+public class DistributionPanel implements IContentProvider {
+    private static class ConfigurationPanelRow extends UIPanel
+    {
+        private static final float textHeight = rowTextHeight_px / rowHeight_px;
+        private static final float textMarginBottom = rowTextMarginBottom_px / rowHeight_px;
+        private static final float textPercentageWidth = 30f / 84f;
+
+        Label lblPercentage;
```

Comment:
You don't need to add those Objects as private (!) fields if they are only used once.